src/app/core/permissions/permission-guard/entity-permission.guard.ts

```diff
@@ -53,6 +53,7 @@ export class EntityPermissionGuard implements CanActivate {
     path = path.replace(/^\//, "");
 
     function isPathMatch(genericPath: string, path: string) {
+      // TODO this does not seem to work with children routes (admin module)
```

Comment:
yes, I was a bit lazy implementing this ... :sweat_smile: 

I think the logic could actually be shared between all guards, so we can make the code from UserRoleGuard reusable:
 https://github.com/Aam-Digital/ndb-core/blob/3b4168b92b9ef214b66e09e594fb77bec2a49a43/src/app/core/permissions/permission-guard/user-role.guard.ts#L75
 
(side note: router.config already contains all the data from config, so parsing the config there is actually unnecessary)

Reply:
I am not sure what you mean here. What is the shared logic? And also regarding the `router.config` I am not fully following. Do you want to extend it here if you have a improvement in mind?

Reply:
generalized the code. Please review my commit :-)